### DISCUSSION TITLE
[Need help] Remove image when use click on btn X with data-toggle="asset-remove" without use event asset:remove

### DISCUSSION THREAD

**Discussion by MinhOmega** [GrapesJS/grapesjs#4876] [Q&A]
Hi author and reader,
I have a problem with a cache of grapesjs so I need get all assets in AssetManager and then call the remove function provided. Here is the code:
```
const existAssets = editor.AssetManager.getAll()
existAssets.forEach((e) => {
  pagebuilderInit.AssetManager.remove(e, { clearCache: true })
})
if (assets.length > 0) {
  setTimeout(() => {
    pagebuilderInit.AssetManager.add(assets)
  }, 1000)
  setTimeout(() => {
    pagebuilderInit.AssetManager.getAll()
  }, 2000)
}
```
That cache has image unknow because i did not add or exist in my code.

So when i need to get image removed by action click btn X of user, it very confuse between auto remove and user action ( both is shown on event `asset:remove`.
I need help for modify that event onRemove with `data-toggle="asset-remove"`. Someone can help me or give another solution for clear cache and removing image.

Note: I use S3 for upload and fetch then show image in FE ( build with NextJS )
Here is image for example:
![image](https://user-images.githubusercontent.com/49482201/215394219-9508a5aa-d848-4485-bfd0-c632b5cd1d4b.png)

Many thank.

**Comment by artf** [comment]
You can remove the asset without triggering the event: `AssetManager.remove(asset, { silent: true })`

**Reply by MinhOmega** [reply]
Thanks, I will try to do it

**Comment by MinhOmega** [comment]
Hi @artf ,
I have tried that solution but I can not know, what `asset` I pass inside the function. You can see the image above, I want to add a function when  I click in X to remove asset both S3 and in AssetManager.

Thank you for your attention

**Comment by MinhOmega** [comment]
[UPDATE]
I have found this docs for handle that. It perfect  and that's what I aspire to do but alr comment. 

https://github.com/GrapesJS/grapesjs/blob/dev/docs/modules/Assets.md?plain=1#L565-L582

```javascript
am.addType('image', {
  // As you adding on top of an already defined type you can avoid indicating
  // `am.getType('image').view.extend({...` the editor will do it by default
  // but you can eventually extend some other type
  view: {
    // If you want to see more methods to extend check out
    // https://github.com/GrapesJS/grapesjs/blob/dev/src/asset_manager/view/AssetImageView.js
    onRemove(e) {
      e.stopPropagation();
      const model = this.model;

      if (confirm('Are you sure?')) {
        model.collection.remove(model);
      }
    }
  },
})
```
